src/DynamoCoreWpf/ViewModels/Core/AnnotationViewModel.cs

```diff
@@ -1406,6 +1414,45 @@ internal bool CanToggleIsVisibleGroup(object parameters)
             return true;
         }
 
+        internal void ToggleIsFrozenGroup(object parameters)
+        {
+            DynamoSelection.Instance.ClearSelection();
+            bool newFrozenState = !this.AnnotationModel.IsFrozen;
+
+            // Collect all nodes inside this group
+            var nodesInGroup = this.AnnotationModel.Nodes.
+                OfType<NodeModel>()
+                .Select(n => n.GUID)
+                .ToList();
+
+            // Collect all nodes inside the nested groups
+            var nestedGroups = this.AnnotationModel.Nodes.OfType<AnnotationModel>();
+            foreach (var nestedGroup in nestedGroups)
+            {
+                nestedGroup.IsFrozen = newFrozenState;
+                nodesInGroup.AddRange(nestedGroup.Nodes.OfType<NodeModel>().Select(n => n.GUID));
+            }
+
+            var command = new DynamoModel.UpdateModelValueCommand(
+                Guid.Empty,
+                nodesInGroup,
+                nameof(this.AnnotationModel.IsFrozen),
+                newFrozenState.ToString());
+
+            this.AnnotationModel.IsFrozen = newFrozenState;
+
+            WorkspaceViewModel.DynamoViewModel.Model.ExecuteCommand(command);
+            WorkspaceViewModel.DynamoViewModel.RaiseCanExecuteUndoRedo();
+            WorkspaceViewModel.HasUnsavedChanges = true;
+
+            Analytics.TrackEvent(Actions.Preview, Categories.GroupOperations, newFrozenState.ToString());
```

Comment:
I don't think we track analytics for freezing nodes, so this can be avoid for freezing groups as well.

Reply:
We do have a `Freeze` Action, maybe use that instead of `Preview`